### PR TITLE
Line.parameter

### PR DESCRIPTION
**Pull request by sn6uv**
This function allows one solve the parametrisation for a point on a line. If there's interest I can implement Line3D too. 

`parameter_of` is probably a better name. Alternatively, `intersection_parameter` which has a more general meaning (planes, lines etc.). The implementation can be cleaned up (duplication with `segment.contains`)

### CODE REVIEW

**Comment by asmeurer** [comment]
This looks good, but I wonder if there's a better name than `parameter`. It took me a bit to understand exactly what it does, and `sympy.geometry.Curve.parameter` already means something else. Maybe `parameter_value`.

**Comment by sn6uv** [comment]
I agree, `parameter` isn't a great name. How about `coordinate(s)` or `coordinates_at`?

The concept could also be extended to anywhere there is a 0D intersection.

**Comment by ChristinaZografou** [comment]
Good quality code. Although, I have to admit it took me some time to completely understand it. Maybe you should add some extra explanation in the docstrings additionally to changing the name of the method.

**Comment by gxyd** [comment]
@sn6uv can you please address the requested changes and also resolve merge conflicts.

File: sympy/geometry/line.py [hunk 1]
```diff
@@ -1158,6 +1158,42 @@ def contains(self, o):
...
+        ``other == t * self.p1 + (1 - t) self.p2`` is True.
+
+        Raises
+        ======
+
+        ValueError
+            When ``other`` is not contained in the line.
+
+        NotImplementedError
+            When ``other`` is not a Point.
```
Comment (gxyd) [inline sympy/geometry/line.py]: double backtick `Point` as well.

**Comment by gxyd** [comment]
I don't think `coordinates_at` is a good name. To me `parameter_value` looks okay. Also if do you use `parameter_value` add a "NOTE" mentioning that this (`parameter_value`) is different from `parameter` in sympy/geometry/curve.py`.

**Comment by gxyd** [comment]
This PR has been inactive for quite sometime, so we are going to close it and open an issue to reference it. We try to keep our pull request queue small and focused on active work.  We encourage you to reopen the pull request if and when you continue to work on this. Please contact us if you have any questions.

Thanks for contributing.

see https://github.com/sympy/sympy/wiki/Closing-pull-requests for our policies on closing pull request.

I've opened https://github.com/sympy/sympy/issues/13515 for this pull request so that we don't forget about it.

**Pull request closed without merging**